internal/guest/runtime/hcsv2/uvm_state.go

```diff
@@ -0,0 +1,91 @@
+//go:build linux
+// +build linux
+
+package hcsv2
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+type rwDevice struct {
+	mountPath string
+	encrypted bool
+	overlays  map[string]struct{}
```

Comment:
How does this get populated? Seems it needs to be protected by the `stateMutex`.

Reply:
nothing is populating it at the moment, but once we start hardening more, we'll be populating this during overlay mount.